#2872 Exposing RBACApiRef  to support simulation on use-cases and to mock API implementation for feature testing.

lavanya-sainik-ericsson opened the pull request:
Exposed ApiRef

Signed-off-by: elavsai lavanya.sainik@ericsson.com

## Hey, I just made a Pull Request!

<!-- Please describe what you added, and add a screenshot if possible.
     That makes it easier to understand the change so we can :shipit: faster. -->

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/community-plugins/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/community-plugins/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

divyanshiGupta commented on workspaces/rbac/.changeset/quiet-lions-float.md:
```diff
@@ -0,0 +1,6 @@
+'@backstage-community/plugin-rbac': minor
+---
+
+#Please enter a summary for your changes.
```
Can you please remove this comment?

lavanya-sainik-ericsson replied:
Thanks @divyanshiGupta for your comment. Have done the change, please have a look and let me know if anything else is needed from my side.

divyanshiGupta commented on workspaces/rbac/.changeset/quiet-lions-float.md:
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage-community/plugin-rbac': minor
```
Please change it to patch

lavanya-sainik-ericsson replied:
Thanks @divyanshiGupta for your comment. Have done the change, please have a look and let me know if anything else is needed from my side.

divyanshiGupta reviewed (changes requested):
@lavanya-sainik-ericsson Please sign your commit and push the changes. Thanks!

lavanya-sainik-ericsson commented:
Hello @divyanshiGupta,

I have fixed the prettier issues in changeset file and also updated the sign off. Please check and let me know if something else is needed. The DCO issue is fixed for the latest patch and failing for previous one but let me know if something else is missing

divyanshiGupta commented:
> Hello @divyanshiGupta,
> 
> I have fixed the prettier issues in changeset file and also updated the sign off. Please check and let me know if something else is needed. The DCO issue is fixed for the latest patch and failing for previous one but let me know if something else is missing

DCO check is still failing. Please sqauash your commits into 1 signed commit and push the changes.

lavanya-sainik-ericsson commented:
Hello @divyanshiGupta ,

Thanks for your comment. I have squashed them and now can see successful run like this 
![image](https://github.com/user-attachments/assets/bbe4cc5b-31c6-41ab-991d-f1bb1825d31c).

Let me know if this is ok to proceed further else I will do a soft rebase and recommit again.

divyanshiGupta commented:
@PatAKnight / @AndrienkoAleksandr can one of you also take a look at this PR and add lgtm? Thanks!

debsmita1 commented:
/lgtm

lavanya-sainik-ericsson commented:
Hello @PatAKnight / @AndrienkoAleksandr / @christoph-jerolimov / @nickboldt 

Can you please have a look and review this PR whenever you have time. 
Thanks @divyanshiGupta & @debsmita1 for your inputs.

 Let me know if any further comments or concerns here 

Thanks

christoph-jerolimov reviewed (changes requested):
Hi @lavanya-sainik-ericsson,

thanks for this contribution. The CI job fails because this is an API change and this must be reflected in the api-report.md.

Please run this command in the workspace to generate a new version of that markdown file, and commit this change as well:

```
yarn tsc && yarn build:api-reports:only
```

Personally, I would also export the RBACBackendClient, but when the apiRef and API interface is enough for your current use case its also fine.

AndrienkoAleksandr commented:
> Can you please have a look and review this PR whenever you have time.

Sorry, I'm not frontend expert...

lavanya-sainik-ericsson commented:
Thanks @christoph-jerolimov for your comment. I have done the suggested changes. Can you please have a look whenever you have time ?
Let me know if any further comments or concerns here

Thanks

lavanya-sainik-ericsson commented:
Thanks @christoph-jerolimov for the comment. I have updated the commit and done the required changes. Kindly review it again whenever you have time and let me know if any further comments.

divyanshiGupta reviewed (approved):
LGTM!

christoph-jerolimov reviewed (approved):
Thanks @lavanya-sainik-ericsson, this looks good. :+1:

Sorry for the delay I was on vacation for almost two weeks. :man_shrugging:

lavanya-sainik-ericsson commented:
Thanks a lot @christoph-jerolimov for the support. 
Hope you had nice vacation :)
